src/main/kotlin/org/rust/ide/inspections/RsUnresolvedReferenceInspection.kt

```diff
@@ -62,6 +62,22 @@ class RsUnresolvedReferenceInspection : RsLocalInspectionTool() {
                 }
             }
 
+            // enum E { A, B }
+            // match e {
+            //     A => {}
+            //     B => {}
+            // }   ^ RsPatBinding
+            override fun visitPatBinding(pat: RsPatBinding) {
+                if (pat.bindingMode != null) return
+                val patIdent = pat.parent as? RsPatIdent ?: return
+                val matchArm = patIdent.parent as? RsMatchArm ?: return
+                if (matchArm.pat != patIdent) return
+                if (pat.reference.multiResolve().isNotEmpty()) return
+
+                val context = AutoImportFix.findApplicableContext(pat)
```

Comment:
It seems the corresponding code doesn't respect expected types because for the following code
```rust
mod inner {
    pub const C: i32 = 0;
}

enum E {
    C
}

fn main() {
    match 0 {
        C => {}
    }
}
```
enum variant is also suggested

<img width="362" alt="image" src="https://user-images.githubusercontent.com/2539310/185897417-dfd13f0c-5446-4f42-9562-23a27af9eda9.png">

Reply:
I propose to do it in another PR because it is not trivial (e.g. we can't just use `isCompatibleTypes` since it returns false for `&E` and `E`)

Reply:
I'd add a test that verifies that intention provides several variants in this case and add a comment that ideally we need to filter out variants

Also, there is simpler example with similar problem:
```rust
mod inner {
    pub const C: i32 = 0;
}

struct Foo;


fn func(foo: Foo) {
    match foo {
        C => {}
    }
}
```

Reply:
Added